### DISCUSSION TITLE
GitHub will make 2FA mandatory

### DISCUSSION THREAD

**Discussion by yan12125** [lxqt/lxqt#2238] [General]
https://github.blog/2022-05-04-software-security-starts-with-the-developer-securing-developer-accounts-with-2fa/

I remember there are some concerns around 2FA (https://github.com/lxqt/lxqt/issues/1852). If enabling 2FA is not an option for some devs, we may need to move away from GitHub.

(It's not quite clear what "contribute code" means. I assume `git push` to repositories on GitHub. There is a request for clarification https://github.com/github/feedback/discussions/16064 but no response yet.)

**Comment by tsujan** [comment]
Frankly, I hate it when Google or Microsoft or... "care" about my security by demanding that I should do this or that. No amount of "professional advice" will convince me to behave paranoically.

EDIT:
My country isn't supported for [SMS authentication](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported). Even if it was, I wouldn't use SMS for git jobs. Using a mobile app is out of question.

**Reply by stefonarch** [reply]
> My country isn't supported for [SMS authentication](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported).

That's an issue. I use this, but it is used only when I don't log in from my "family PC" for some days or week. 
On the laptop the only thing I see is sometimes the question if numbers and reset emails ecc are up-to-date.



> Using a mobile app is out of question.

I don't have and want a smartphone, but for the 2 apps I need sometimes I've setup an VM with genymotion and a android phone - it works well.

**Reply by stefonarch** [reply]
Migration is probably not a solution too, as 2FA sneaks in everywhere.

**Reply by yan12125** [reply]
> Using a mobile app is out of question.

FWIW, personally I use the TOTP feature from KeePassXC. With some tricks like converting QR codes to hex secrets, it works with almost all websites that requires/suggests 2FA. It seems there are many more TOTP applications available for GNU/Linux: https://www.reddit.com/r/linuxquestions/comments/90lkk3/twofactor_authentication_client_for_gnulinux/.

**Reply by tsujan** [reply]
> Migration is probably not a solution too, as 2FA sneaks in everywhere.

Who knows what will happen until 2024? But if GitHub still insists on this, I won't use it. I don't think GitLab will make 2FA obligatory too; if it does, I won't use it either. There's a limit to this mobile-app nonsense.

They themselves say, "only approximately 16.5% of active GitHub users... use one or more forms of 2FA", but they don't ask why. Don't so many devs know they're "insecure"?

**Reply by stefonarch** [reply]
In 2024 we are all hacked ;)
I didn't think about the TOTP, that's easy too and  no mobile apps involved.

**Comment by yan12125** [comment]
GitHub pushes the deadline earlier for some developers. By Mar 2023, "users who created a release", "users who are Enterprise and Organization administrators", and some others will be required to enable 2FA.

I assume @tsujan will need 2FA to get LXQt 1.3 released on GitHub.

https://github.blog/2022-12-14-raising-the-bar-for-software-security-next-steps-for-github-com-2fa/

**Reply by tsujan** [reply]
Then I might need to learn your KeePassXC trick because, frankly, GitHub's suggestion about using mobile phones is absolutely ridiculous, if not an insult, to me — if it weren't for LXQt, I'd migrate to GitLab today.

**Reply by Chiitoo** [reply]
Alternatively, 'oathtool --totp --base32' I feel makes it a rather simple endeavour.

**Reply by tsujan** [reply]
@Chiitoo, thanks a lot! I made a test GitHub account, enabled 2FA in it, and used `oathtool` without any hassle. No need to a smartphone beside the laptop :)

Will enable 2FA on my main account when GitHub makes it mandatory.

**Comment by tsujan** [comment]
BTW, `git push` is done by using an "access token". While I don't have 2FA yet, it's about a year that I use personal access token with command line — because GitHub made me do so last year, by threatening me ;)

And with `oathtool`, 2FA's annoyance lasts only ~10 seconds each time. So, unless GitHub makes yet another layer of "protection" mandatory, it's tolerable.

**Reply by tsujan** [reply]
Enabled 2FA yesterday. Leaving this comment with it...
